publications/serializers.py

```diff
@@ -1,21 +1,50 @@
 """publications serializers."""
 
 from rest_framework_gis import serializers
-from .models import Publication
+from rest_framework import serializers as drf_serializers
+from .models import Publication, Subscription, Journal
 from django.contrib.auth import get_user_model
-User = get_user_model()
 
-from publications.models import Publication,Subscription
-from django.contrib.auth import get_user_model
 User = get_user_model()
 
+class JournalSerializer(drf_serializers.ModelSerializer):
+    """Serializer for Journal model."""
+
+    class Meta:
+        model = Journal
+        fields = [
+            "id",
+            "name",
+            "issn_l",
+            "openalex_id",
+            "openalex_url",
+            "publisher_name",
+            "works_count",
+            "works_api_url",
+        ]
+
 class PublicationSerializer(serializers.GeoFeatureModelSerializer):
     """publication GeoJSON serializer."""
+    source_details = JournalSerializer(source="source", read_only=True)
 
     class Meta:
-        """publication serializer meta class."""
         model = Publication
-        fields = ("id", "title" ,"abstract", "publicationDate", "url", "doi", "creationDate", "lastUpdate", "timeperiod_startdate", "timeperiod_enddate")
+        fields = (
+            "id",
+            "title",
+            "abstract",
+            "publicationDate",
+            "url",
+            "doi",
+            "creationDate",
+            "lastUpdate",
+            "timeperiod_startdate",
+            "timeperiod_enddate",
+            "source",
+            "source_details",       
```

Comment:
In the serialisation of the publication, we do _not_ want all fields from the journal - it's just blowing things up. Instead we want `id` (so that a client can retrieve the detailed information from our API), `name` and `issn_l` so that we can render the name nicely, and something that is a URL, so let's use `openalex_url` which we can then turn into a link in the UI.